cupy/core/_scalar.pxd

```diff
@@ -13,12 +13,23 @@ cdef class CScalar(CPointer):
         char kind
         int8_t size
 
+    @staticmethod
+    cdef CScalar from_int32(int32_t value)
+
+    @staticmethod
+    cdef CScalar from_numpy_scalar_with_dtype(object x, object dtype)
```

Comment:
Note:
These methods are still considered as private because the class `CScalar` is in a private module and Cython "cannot call a static method on an instance variable".
https://github.com/cython/cython/blob/3dc2b9dfc23638fbef2558d619709b5235d5df08/Cython/Compiler/ExprNodes.py#L5619-L5624